src/general.rs

```diff
@@ -249,16 +250,23 @@ pub fn initialize(
                     token_types: ctx
                         .config
                         .semantic_tokens
-                        .keys()
+                        .iter()
                         .cloned()
-                        .map(|x| x.into())
+                        .map(|token| token.name.into())
+                        // Collect into HashSet first to remove duplicates
+                        .collect::<HashSet<SemanticTokenType>>()
+                        .into_iter()
                         .collect(),
                     token_modifiers: ctx
                         .config
-                        .semantic_token_modifiers
-                        .keys()
+                        .semantic_tokens
+                        .iter()
                         .cloned()
-                        .map(|x| x.into())
+                        // Get all modifiers used in token definitions
+                        .flat_map(|token| token.modifiers)
+                        // Collect into HashSet first to remove duplicates
+                        .collect::<HashSet<SemanticTokenModifier>>()
+                        .into_iter()
                         .collect(),
                     formats: vec![TokenFormat::RELATIVE],
                     overlapping_token_support: None,
```

Comment:
We still send `semantic_tokens: None,` in `src/general.rs:86` as part of the `WorkspaceClientCapabilities`.

Do we already support that? I didn't check what difference it will make in practice to set that field.

Reply:
I don't think so. The only value that `SemanticTokensWorkspaceClientCapabilities` includes currently is a flag about supporting the request `workspace/semanticTokens/refresh`.

As far as I understand that request, we would need to proxy it to Kakoune and have the editor respond with `textDocument/semanticTokens/full` for all open files.